lib/associations/belongs-to-many.js

```diff
@@ -442,9 +442,11 @@ class BelongsToMany extends Association {
     const model = association.target;
     const sequelize = model.sequelize;
 
+    const primaryKeyField = model.attributes[model.primaryKeyAttribute].field || model.primaryKeyAttribute;
+
     options = Utils.cloneDeep(options);
     options.attributes = [
-      [sequelize.fn('COUNT', sequelize.col([association.target.name, model.primaryKeyAttribute].join('.'))), 'count']
+      [sequelize.fn('COUNT', sequelize.col([association.target.name, primaryKeyField].join('.'))), 'count']
```

Comment:
use `model.primaryKeyField` instead

Reply:
Not sure how I missed that property :disappointed: